runbotics-orchestrator/src/main/java/com/runbotics/service/impl/BotServiceImpl.java

```diff
@@ -90,13 +94,24 @@ public void delete(Long id) {
         botRepository.deleteById(id);
     }
 
-    @Override
-    public Page<BotDTO> getBotsWithCollection(Pageable pageable, List<String> collectionsNames, String login) {
-        if (collectionsNames.isEmpty()) {
-            return botRepository.getAllAvailableToCurrentUser(pageable, login)
+    public Page<BotDTO> getBotPageForUser(BotCriteria criteria, Pageable page, User user) {
+        List<String> commonCollections = getCommonCollections();
+
+        if (criteria.getCollection() != null) {
+            List<String> names = criteria.getCollection().getIn();
+            return this.botRepository
+                .findAllByUserAndByCollectionNames(page, user.getId(), names, commonCollections)
                 .map(botMapper::toDto);
         }
-        return botRepository.getAllAvailableToCurrentUserWithTags(pageable, login, collectionsNames)
+        return this.botRepository
+            .findAllByUser(page, user.getId(), commonCollections)
             .map(botMapper::toDto);
     }
+
+    private List<String> getCommonCollections() {
+        return Arrays.asList(
+            BotCollectionConstants.PUBLIC_COLLECTION,
+            BotCollectionConstants.GUEST_COLLECTION
+        );
+    }
```

Comment:
duplication from `BotCollectionServiceImpl.java`, transfer it to some utils and reuse